pkg/controller/llmisvc/webhook/llminferenceservice_validator.go

```diff
@@ -171,3 +170,97 @@ func (l *LLMInferenceServiceValidator) validateRouterCrossFieldConstraints(llmSv
 
 	return allErrs
 }
+
+func (l *LLMInferenceServiceValidator) validateParallelismConstraints(llmSvc *v1alpha1.LLMInferenceService) field.ErrorList {
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, l.validateWorkloadParallelism(field.NewPath("spec"), &llmSvc.Spec.WorkloadSpec)...)
+
+	if llmSvc.Spec.Prefill != nil {
+		allErrs = append(allErrs, l.validateWorkloadParallelism(field.NewPath("spec").Child("prefill"), llmSvc.Spec.Prefill)...)
+	}
+
+	return allErrs
+}
+
+func (l *LLMInferenceServiceValidator) validateWorkloadParallelism(basePath *field.Path, workload *v1alpha1.WorkloadSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if workload.Worker != nil && workload.Parallelism == nil {
+		allErrs = append(allErrs, field.Invalid(
+			basePath.Child("worker"),
+			workload.Worker,
+			"when worker is specified, parallelism must be configured for either data parallelism or pipeline parallelism",
+		))
+		return allErrs
+	}
+
+	if workload.Parallelism == nil {
+		return field.ErrorList{}
+	}
+
+	parallelismPath := basePath.Child("parallelism")
+	parallelism := workload.Parallelism
+
+	// setting both Pipeline and any of DataLocal or Data is rejected
+	if parallelism.Pipeline != nil && (parallelism.Data != nil || parallelism.DataLocal != nil) {
```

Comment:
can't we use `IsDataParallel() && IsPipelineParallel()` ?

Reply:
even if they check for > 0 in the method, I guess it will be error-ing out anyway with the other check below.

Reply:
that makes it simpler, thanks for the suggestion